### ISSUE TITLE
'Sign up' dialog

### ISSUE THREAD

**Issue by Yasinaaa** [Zakyat/zakyat-frontend#40]
In Figma - Sign Up
<img width="480" alt="Screenshot 2020-06-09 at 08 37 34" src="https://user-images.githubusercontent.com/10158025/84134235-f1db2d00-aa50-11ea-8f81-bb3264509252.png">

When invalid email was input
In Figma - Sign Up. Week Password
<img width="480" alt="Screenshot 2020-06-09 at 08 37 43" src="https://user-images.githubusercontent.com/10158025/84134236-f1db2d00-aa50-11ea-8d90-dbfc4f2edb1c.png">

When forgot password 'Забыли пароль' click, opens view for email input
In Figma - Sign up. Average password
<img width="477" alt="Screenshot 2020-06-09 at 08 37 54" src="https://user-images.githubusercontent.com/10158025/84134238-f273c380-aa50-11ea-8adf-6a967236eea3.png">

When letter for changing the password was sent
In Figma - Success
<img width="471" alt="Screenshot 2020-06-09 at 08 38 03" src="https://user-images.githubusercontent.com/10158025/84134239-f273c380-aa50-11ea-82ac-29b90aec20f2.png">

When week password was input
In Figma - Sign Up. Week Password
<img width="483" alt="Screenshot 2020-06-09 at 08 37 01" src="https://user-images.githubusercontent.com/10158025/84134229-f0aa0000-aa50-11ea-9734-a1ad01ecf0ce.png">

When average password was input
In Figma - Sign Up. Average Password
<img width="488" alt="Screenshot 2020-06-09 at 08 37 08" src="https://user-images.githubusercontent.com/10158025/84134230-f0aa0000-aa50-11ea-9187-0036a63df791.png">

When strong password was input
In Figma - Sign Up. Good Password
<img width="484" alt="Screenshot 2020-06-09 at 08 37 16" src="https://user-images.githubusercontent.com/10158025/84134231-f1429680-aa50-11ea-875f-03fb2008f54b.png">

**Comment by chackydude** [comment]
I am planning to realise following steps to solve the issue:
1) Create **sign-up** directory into the **components** directory
2) Add two types of components: small and reusable (**handy-components** directory) and main componentns (**main-components** directory)
3) Move **SignUp.vue**, **SignIn.vue**, **RecoveryPass.vue** and **Success.vue** into the **main-components** directory
4) Move **Email.vue** and **Password.vue** into the **handy-components** directory

Example of file structure:
**root**
|--components
|----sign-up
|------main-components
|---------SignUp.vue
|---------SignIn.vue
|---------RecoveryPass.vue
|---------Success.vue
|------handy-components
|---------Email.vue
|---------Password.vue

5) The logic of verification of password and email I will put into the same componentns: **Password.vue** and **Email.vue**
6) Links to the registration with using social networks, checkboxes and other I will put into the main components: **SignUp.vue**, **SignIn.vue**, **RecoveryPass.vue**

**Comment by aabounegm** [comment]
The convention we are following is that every folder in the **components** directory is associated with a `.vue` file in **pages** that imports those components and combines them to form the page. Components that are common to all pages (Header, Footer, Nav bar, ...) reside in the root of the **components** directory. The sign up/in components do not really have a page of their own, they are common to the layout as a whole. If you think the root will get over-crowded, we can move them all to a **main** or **common** folder (repeating the word "-components" is really redundant, tbh).
Also, I don't think email and password really need a separate component. They can be implemented in about 5 lines max with [`<v-text-field>`](https://vuetifyjs.com/en/components/text-fields/).

**Comment by chackydude** [comment]
Ok, did I understand right that you recommend to reorganize file structure as follows?

**root**
|--components
|-------SignUp.vue (that's component is already partially written by someone and named like RegistrationDialog.vue)
|-------SignIn.vue
|-------RecoveryPass.vue
|-------Success.vue

or something like that: 

**root**
|--components
|----common
|-------SignUp.vue
|-------SignIn.vue
|-------RecoveryPass.vue
|-------Success.vue

Also I can rename components like SingInDialog.vue, SucessDialog.vue etc. for common name-style.

**Comment by aabounegm** [comment]
Yes, exactly. If you choose the 2nd variant, move other components that are already in the root as well. Feel free to use nested folders for further organization (I would prefer if the nested component does not get imported by the page directly, but by another component that composes them together). This will probably become more relevant when you start implementing, because the way I imagine it is that there is one component whose root is a [`<v-dialog>`](https://vuetifyjs.com/en/components/dialogs/) and its content will be one of the 4 components you mentioned with a [`v-tab-items`](https://vuetifyjs.com/en/components/tabs/#api) that selects from them

**Comment by chackydude** [comment]
I added pull request of **sign_up_dialog** branch, please check. Now I have one problem with styling Vuetify subcomponent **label** of **<v-checkbox>**: class-styling like a **class="text-12"**, styling with using **.v-checkbox label {fon-size: 12px}** or **.v-label {font-size: 12px}** doesn't work and I still thinking how to solve that this. Do u have any ideas?)

**Comment by aabounegm** [comment]
I'm not exactly sure what you are talking about, but let's move the discussion to the PR itself, where you can highlight the lines of code in question
